contrib/fparser/util/bytecoderules_parser.cc

```diff
@@ -1269,7 +1269,7 @@ namespace
             }
 
             Node* head = &global_head;
-            for(size_t b=sequence.size(); b-->0; )
+            for(ptrdiff_t b=sequence.size(); b-->0; )
```

Comment:
How was this working before with an unsigned type and this cutesy `b-->0` BS? When `b==0` the test fails and then you post-decrement so the number wraps to max size_t, but we never see it because the loop is done?

Reply:
Precisely.  It's apparently a popular idiom, too; this is what `-fsanitize=integer` was complaining about in a couple places in libstdc++ `basic_string` code.